### ISSUE TITLE
js-reporter does not recognize filtered test points when using QUnit.config.filter

### ISSUE THREAD

**Issue by muthu90ec** [qunitjs/qunit#1407]
## Tell us about your runtime:

* **QUnit version**: 2.9.1
* **What environment are you running QUnit in? (e.g., browser, Node)**: node
* **How are you running QUnit? (e.g., script, testem, Grunt)**: qunit cli

## What are you trying to do?
[js-reporter](https://github.com/js-reporters/js-reporters#readme) errors out when QUnit.config.filter is used. QUnit.config.filter works fine, but the printing portion errors out when using js-reporter.

**Comment by trentmwillis** [comment]
Can you provide a reproduction or, at least, a sample console output of what error is occurring? That would make it easier for us to debug and fix.

**Comment by muthu90ec** [comment]
> Can you provide a reproduction or, at least, a sample console output of what error is occurring? That would make it easier for us to debug and fix.
> 
> Related issue from js-reporters project: [js-reporters/js-reporters#113](https://github.com/js-reporters/js-reporters/issues/113). Unsure where the fix will be needed, so leaving both issues open for now.

Here is the test tExample.js:

```
let bpPoint = [': test try bp']
const regexEndAppender = val => "^" + val + "$";
let mapedP = bpPoint.map(regexEndAppender)
QUnit.config.filter = `!/${mapedP.join("|")}/`
var add = require(require('path').resolve('addFunc.js'));
QUnit.test("test try bp", function (assert) {
    assert.ok(add(1,2) === 3, "dummy 1 of 2");
    assert.ok(false, "first test");
});
QUnit.test("test 2", function (assert) {
    assert.ok(add(1,2) === 3, "dummy 1 of 2");
    assert.ok(true, "second test");
});
```
using a simple custom js-reporter as customRep.js:
```
var JsReporters = require('js-reporters');
var runner = JsReporters.autoRegister();
var path = require('path');

runner.on('testEnd', function(test) {
    //Do some stuff
});

runner.on('runStart', function(test) {
   //Do some stuff
});


runner.on('runEnd', function(test) {
    //Do some stuff
});
```

Command used to run:
```
qunit -require customRep.js tExample.js

```

Error produced:

```
TAP version 13
ok 1 test 2
1..1
# pass 1
# skip 0
# todo 0
# fail 0
not ok 2 global failure
  ---
  message: "Cannot read property 'slice' of undefined"
  severity: failed
  actual: {}
  expected: undefined
  stack: TypeError: Cannot read property 'slice' of undefined
    at createTestEnd 
  ...

```

**Comment by trentmwillis** [comment]
Thanks for the reproduction!

**Comment by Krinkle** [comment]
I suspect this is caused due to https://github.com/qunitjs/qunit/issues/1416 which I'm reviewing a fix for and should be out in a week or two in the next release.

**Comment by smcclure15** [comment]
This still reproduces in QUnit 2.13.0 - the references here have been a little circuitous, so I don't want them to get lost.
Here's some smaller repros:

```
var jsReporters = require('js-reporters');
jsReporters.autoRegister();
```
```
QUnit.config.filter = '/foo/';

QUnit.test('foo', assert => {
    assert.ok(1);
});
QUnit.test('bar', assert => {
    assert.ok(1);
});
```

The error is coming from the QUnitAdapter:
```
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'slice' of undefined
    at createTestEnd (/SNIP/node_modules/js-reporters/dist/js-reporters.js:1010:111)
    at /SNIP/node_modules/js-reporters/dist/js-reporters.js:1163:32
    at Array.forEach (<anonymous>)
    at QUnitAdapter.emitData (/SNIP/node_modules/js-reporters/dist/js-reporters.js:1161:19)
    at QUnitAdapter.onDone (/SNIP/node_modules/js-reporters/dist/js-reporters.js:1201:12)
    at /SNIP/node_modules/qunit/qunit/qunit.js:2546:33
```

and directly attaching QUnit as the runner does avoid this:
```
jsReporters.TapReporter.init(runner);
```

With fixes in the 2.13.0 release, I'm thinking this looks like a js-reporters bug.
